### PR TITLE
Exit error code from provision.sh

### PR DESCRIPTION
**Pull request by bowang-666**
Correctly exit error code that comes from provision.sh.

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_cmd.sh [hunk 1]
```diff
@@ -67,4 +67,4 @@ SGX_NODE={{GetVariable "sgxNode"}}
 AUDITD_ENABLED={{GetVariable "auditdEnabled"}} 
 CONFIG_GPU_DRIVER_IF_NEEDED={{GetVariable "configGPUDriverIfNeeded"}}
 ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED={{GetVariable "enableGPUDevicePluginIfNeeded"}}
-/usr/bin/nohup /bin/bash -c "/bin/bash /opt/azure/containers/provision.sh >> /var/log/azure/cluster-provision.log 2>&1; systemctl --no-pager -l status kubelet 2>&1 | head -n 100"
\ No newline at end of file
+/usr/bin/nohup /bin/bash -c "/bin/bash /opt/azure/containers/provision.sh >> /var/log/azure/cluster-provision.log 2>&1 || exit $?; systemctl --no-pager -l status kubelet 2>&1 | head -n 100"
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_cmd.sh:70]: do we want to show kubelet status regardless of the exit status code?

Comment (bowang-666) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh]: updated to exit code in end of script

File: parts/linux/cloud-init/artifacts/cse_cmd.sh [hunk 2]
```diff
@@ -67,4 +67,4 @@ SGX_NODE={{GetVariable "sgxNode"}}
 AUDITD_ENABLED={{GetVariable "auditdEnabled"}} 
 CONFIG_GPU_DRIVER_IF_NEEDED={{GetVariable "configGPUDriverIfNeeded"}}
 ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED={{GetVariable "enableGPUDevicePluginIfNeeded"}}
-/usr/bin/nohup /bin/bash -c "/bin/bash /opt/azure/containers/provision.sh >> /var/log/azure/cluster-provision.log 2>&1; systemctl --no-pager -l status kubelet 2>&1 | head -n 100"
\ No newline at end of file
+/usr/bin/nohup /bin/bash -c "/bin/bash /opt/azure/containers/provision.sh >> /var/log/azure/cluster-provision.log 2>&1 || PROVISION_CODE=$?; systemctl --no-pager -l status kubelet 2>&1 | head -n 100; exit ${PROVISION_CODE}"
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_cmd.sh:70]: && PROVISION_CODE=$?

we always want to get the code

Comment (bowang-666) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh]: updated

**Pull request merged**